pkg/operator/controller/ingress/status.go

```diff
@@ -821,6 +824,12 @@ func IngressStatusesEqual(a, b operatorv1.IngressControllerStatus) bool {
 				}
 			}
 		}
+		if a.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil && b.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+			a.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack != nil && b.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack != nil {
+			if a.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack.FloatingIP != b.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack.FloatingIP {
+				return false
```

Comment:
Hm, this logic does not seem correct.  Try adding the following test case in `Test_IngressStatusesEqual`:
```go
		{
			description: "OpenStack providerParameters present for a but nil for b",
			expected:    false,
			a:           icStatusWithFloatingIP("1.2.3.4"),
			b: operatorv1.IngressControllerStatus{
				EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
					Type: operatorv1.LoadBalancerServiceStrategyType,
					LoadBalancer: &operatorv1.LoadBalancerStrategy{
						ProviderParameters: nil,
					},
				},
			},
		},
```
This test case should pass, but it fails.

Reply:
indeed, I fixed it.